eng/common/scripts/get-codeowners.ps1

```diff
@@ -1,49 +1,50 @@
+  [string]$CodeOwnerFileLocation = "../../../.github/CODEOWNERS", # The place we fetch CODEOWNERS file
+  [string]$VsoVariable = "" # Option of write code owners into devop variable
 )
-$target = $TargetDirectory.ToLower().Trim("/")
-$codeOwnersLocation = Join-Path $RootDirectory -ChildPath ".github/CODEOWNERS"
-$ownedFolders = @{}
 
-if (!(Test-Path $codeOwnersLocation)) {
-  Write-Host "Unable to find CODEOWNERS file in target directory $RootDirectory"
-  exit 1
+# Check if the retrieve-codeowners tool exsit or not.
+function InstallRetrieveCodeOwnersTool() {
+  if (Get-Command "$ToolPath/retrieve-code-owners") {
+    return
+  }
+  Write-Warning "Installing the retrieve-code-owner tool under $ToolPath... "
+  dotnet tool install --tool-path $toolPath --add-source "$DevOpsFeed" --version "$ToolVersion" "Azure.Sdk.Tools.RetrieveCodeOwners"
 }
 
-$codeOwnersContent = Get-Content $codeOwnersLocation
-
-foreach ($contentLine in $codeOwnersContent) {
-  if (-not $contentLine.StartsWith("#") -and $contentLine){
-    $splitLine = $contentLine -split "\s+"
-    
-    # CODEOWNERS file can also have labels present after the owner aliases
-    # gh aliases start with @ in codeowners. don't pass on to API calls
-    $ownedFolders[$splitLine[0].ToLower().Trim("/")] = ($splitLine[1..$($splitLine.Length)] `
-      | ? { $_.StartsWith("@") } `
-      | % { return $_.substring(1) }) -join ","
+# Get the json property from tool command.
+function GetCodeOwnersEntryFromCommand() {
+  if (Get-Command "$ToolPath/retrieve-code-owners") {
+    Write-Error "The retieve-code-owners tool is not properly installed. Please check your tool path. $ToolPath"
   }
+  return & "$ToolPath/retrieve-code-owners" --target-directory "$CodeOwnerPathExpression" --code-owner-file-path "$CodeOwnerFileLocation" 
 }
 
-$results = $ownedFolders[$target]
-
-if ($results) {
-  Write-Host "Found a folder $results to match $target"
+function GetCodeOwners([string]$codeOwnersString) 
+{ 
+  $codeOwnersJson = $codeOwnersString | ConvertFrom-Json
+  if (!$codeOwnersJson) {
+    Write-Host "No code owners returned from the path: $CodeOwnerPathExpression"
+    return ""
+  }
   
+  $codeOwners = $codeOwnersJson.Owners -join ","
+  Write-Host "Code owners are $codeOwners"
   if ($VsoVariable) {
-    $alreadyPresent = [System.Environment]::GetEnvironmentVariable($VsoVariable)
-
-    if ($alreadyPresent) { 
-      $results += ",$alreadyPresent"
-    }
-    Write-Host "##vso[task.setvariable variable=$VsoVariable;]$results"
+    Write-Host "##vso[task.setvariable variable=$VsoVariable;]$codeOwners"
   }
-
-  return $results
+  return $codeOwners
 }
-else {
-  Write-Host "Unable to match path $target in CODEOWNERS file located at $codeOwnersLocation."
-  Write-Host ($ownedFolders | ConvertTo-Json)
+
+InstallRetrieveCodeOwnersTool
+
+$codeOwnerToolOutput = GetCodeOwnersEntryFromCommand
+# Failed at the command of fetching code owners.
+if ($LASTEXITCODE -ne 0) {
   return ""
```

Comment:
We should probably return an empty list.

Reply:
Why don't we return string with ","
I feel like string is much easy to deliver between yaml and scripts. It is also easy to parse.

Reply:
I think to keep both the object format and the yaml<-->script format, this could be a JSON stringified empty list, i.e. `"[]"` or `@() | ConvertTo-Json`.